core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2.scala

```diff
@@ -57,25 +66,19 @@ trait BillingApiServiceV2 extends UserInfoDirectives {
           pathEndOrSingleSlash {
             get {
               parameters(
-                "startDate".as[String],
-                "endDate".as[String],
+                "startDate".as[DateTime],
+                "endDate".as[DateTime],
                 "aggregationKey"
-                  .as[SpendReportingAggregationKey](spendReportingAggregationKeyUnmarshaller)
-                  .?)
-              { (startDate, endDate, aggregationKey) =>
+                  .as[SpendReportingAggregationKeyWithSub](aggregationKeyParameterUnmarshaller)
+                  .repeated)
+              { (startDate, endDate, aggregationKeyParameters) =>
                 complete {
-                  Try {
-                    (DateTime.parse(startDate), DateTime.parse(endDate))
-                  }.recover {
-                    case _: IllegalArgumentException => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "invalid dates provided"))
-                  }.map { case (parsedStartDate, parsedEndDate) =>
-                    spendReportingConstructor(userInfo).getSpendForBillingProject(
-                      RawlsBillingProjectName(projectId),
-                      parsedStartDate,
-                      parsedEndDate.plusDays(1).minusMillis(1),
-                      aggregationKey
-                    )
-                  }
+                  spendReportingConstructor(userInfo).getSpendForBillingProject(
```

Comment:
It looks like the changes in this file aren't covered by tests. Are there unit tests that verify the routing based on URL that could be extended to cover this code?